tests/unit/test_network_connections.py

```diff
@@ -2475,6 +2476,74 @@ def test_full_state_absent_defaults(self):
             },
         )
 
+    def _test_ethtool_changes(self, input_features, expected_features):
+        """
+        When passing a dictionary 'input_features' each feature and their value
+        to change, and a dictionary 'expected_features' with the expected result
+        in the configuration, the expected and resulting connection are created
+        and validated.
+        """
+        custom_ethtool_features = copy.deepcopy(ETHTOOL_FEATURES_DEFAULTS)
+        custom_ethtool_features.update(expected_features)
+        expected_ethtool = {"features": custom_ethtool_features}
+        input_connection = {
+            "ethtool": {"features": input_features},
+            "name": "5",
+            "persistent_state": "present",
+            "type": "ethernet",
+        }
+
+        expected_connection = {
+            "actions": ["present"],
+            "ethtool": expected_ethtool,
+            "interface_name": "5",
+            "persistent_state": "present",
+            "state": None,
+            "type": "ethernet",
+        }
+        self.check_one_connection_with_defaults(input_connection, expected_connection)
+
+    def test_set_ethtool_feature(self):
+        """
+        When passing the name of an non-deprecated ethtool feature, their
+        current version is updated.
+        """
+        input_features = {"tx_tcp_segmentation": "yes"}
+        expected_feature_changes = {"tx_tcp_segmentation": True}
+        self._test_ethtool_changes(input_features, expected_feature_changes)
+
+    def test_set_deprecated_ethtool_feature(self):
+        """
+        When passing a deprecated name, their current version is updated.
+        """
+        input_features = {"tx-tcp-segmentation": "yes"}
+        expected_feature_changes = {"tx_tcp_segmentation": True}
+        self._test_ethtool_changes(input_features, expected_feature_changes)
+
+    def test_invalid_ethtool_settings(self):
+        """
+        When both the deprecated and current version of a feature are stated,
+        a Validation Error is raised.
+        """
+        input_features = {"tx-tcp-segmentation": "yes", "tx_tcp_segmentation": "yes"}
+        features_validator = (
+            network_lsr.argument_validator.ArgValidator_DictEthtoolFeatures()
+        )
+        self.assertValidationError(features_validator, input_features)
+
+    def test_deprecated_ethtool_names(self):
+        """
+        Test that for each validator in
+        ArgValidator_DictEthtoolFeatures.nested there is another non-deprecated
+        validator that has the name from the deprecated_by attribute"
+        """
+        validators = (
+            network_lsr.argument_validator.ArgValidator_DictEthtoolFeatures().nested
+        )
+        for name, validator in validators.items():
+            if type(validator) is network_lsr.argument_validator.ArgValidatorDeprecated:
```

Comment:
You are using isinstance in the other code and it seems to be correct here, too:

```suggestion
            if isinstance(validator, network_lsr.argument_validator.ArgValidatorDeprecated):
```